File: java-api-core/test/com/kinvey/java/auth/KinveyAuthRequestTest.java
package com.kinvey.java.auth;

import com.kinvey.java.core.KinveyHeaders;
import com.kinvey.java.core.KinveyMockUnitTest;
import com.kinvey.java.dto.User;
import com.kinvey.java.store.UserStoreRequestManager;
import com.kinvey.java.testing.MockKinveyAuthRequest;

import java.lang.reflect.Field;

/**
 * Created by edward on 10/13/15.
 */
public class KinveyAuthRequestTest extends KinveyMockUnitTest {

    private UserStoreRequestManager currentUser;

    private void initializeUser() {
        currentUser = new UserStoreRequestManager(getClient(), new MockKinveyAuthRequest.MockBuilder(getClient().getRequestFactory().getTransport(),
                getClient().getJsonFactory(), "mockAppKey","mockAppSecret",null));
    }

    public void testHeaders() {
        initializeUser();
        try {
            UserStoreRequestManager.LoginRequest login = currentUser.loginBlocking();

            login.buildAuthRequest();

            Field f = UserStoreRequestManager.LoginRequest.class.getDeclaredField("request");
            f.setAccessible(true);
            KinveyAuthRequest authRequest = (KinveyAuthRequest) f.get(login);

            f = KinveyAuthRequest.class.getDeclaredField("kinveyHeaders");
            f.setAccessible(true);
            KinveyHeaders requestHeaders = (KinveyHeaders) f.get(authRequest);

            assertFalse(requestHeaders.containsKey("Authorization"));

        }catch (Exception e){
            assertNull(e);
//            assertFalse("Shouldn't have thrown an exception", true);
        }

    }


}

File: java-api-core/test/com/kinvey/java/auth/KinveyAuthRequestTest.kt
package com.kinvey.java.auth

import com.kinvey.java.core.KinveyHeaders
import com.kinvey.java.core.KinveyMockUnitTest
import com.kinvey.java.dto.BaseUser
import com.kinvey.java.store.UserStoreRequestManager
import com.kinvey.java.store.UserStoreRequestManager.LoginRequest
import com.kinvey.java.testing.MockKinveyAuthRequest.MockBuilder
import junit.framework.TestCase
import java.lang.reflect.Field

/**
 * Created by edward on 10/13/15.
 */
class KinveyAuthRequestTest : KinveyMockUnitTest<BaseUser>() {

    private var currentUser: UserStoreRequestManager<*>? = null

    private fun initializeUser() {
        currentUser = UserStoreRequestManager(client, MockBuilder(client?.requestFactory?.transport,
            client?.jsonFactory, "mockAppKey", "mockAppSecret", null) as KinveyAuthRequest.Builder<BaseUser>)
    }

    fun testHeaders() {
        initializeUser()
        try {
            val login = currentUser?.loginBlocking()
            login?.buildAuthRequest()
            var f: Field? = LoginRequest::class.java.getDeclaredField("request")
            f?.isAccessible = true
            val authRequest = f?.get(login) as KinveyAuthRequest<*>
            f = KinveyAuthRequest::class.java.getDeclaredField("kinveyHeaders")
            f.isAccessible = true
            val requestHeaders = f.get(authRequest) as KinveyHeaders
            TestCase.assertFalse(requestHeaders.containsKey("Authorization"))
        } catch (e: Exception) {
            TestCase.assertNull(e)
//          assertFalse("Shouldn't have thrown an exception", true);
        }
    }
}